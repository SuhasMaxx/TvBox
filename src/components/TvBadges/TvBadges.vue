<template>
  <div class="color-info-blocks">
    <label v-if="showDetails.rating.average" class="rating" title="Ratings"
      >â˜… {{ showDetails.rating.average }}</label
    >
    <label v-if="showDetails.premiered" class="year" title="Premiered on">{{
      premierYear
    }}</label>
    <label v-if="showDetails.language" class="language" title="Language">{{
      showDetails.language
    }}</label>
    <label v-if="showDetails.runtime" class="runtime" title="Runtime"
      >{{ showDetails.runtime }} mins</label
    >
  </div>
</template>
<script>
import { UtilsMixins } from "../../utils/utils.mixin.js";
export default {
  mixins: [UtilsMixins],
  props: {
    showDetails: {
      type: Object,
      default: null,
    },
  },
  computed: {
    premierYear: function () {
      return this.getYear(this.showDetails.premiered);
    },
  },
};
</script>
<style scoped>
.color-info-blocks {
  display: block;
  font-style: italic;
}
.color-info-blocks label:nth-child(1) {
  margin-left: 0em;
}
.color-info-blocks label {
  padding: 0.5em;
  border-radius: 1em;
  margin-left: 0.5em;

  color: var(--cultured);
  min-width: 25px;
  font-size: 0.8em;
  display: inline-block;
}
.rating {
  background-color: var(--carnelian);
}
.year {
  background-color: #1bba18;
}
.language {
  background-color: #7f1b18;
}
.runtime {
  background-color: #187f1b;
}
</style>
