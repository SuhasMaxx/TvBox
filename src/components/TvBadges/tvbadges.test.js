import TvBadges from "./TvBadges.vue";
import { shallowMount } from "@vue/test-utils";
import showDetails from "../../utils/showDetails.mock.js";

describe("TvBadges.vue", () => {
  it("render the component", () => {
    const wrapper = shallowMount(TvBadges, {
      props: {
        showDetails: showDetails.singleShowDetails,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
