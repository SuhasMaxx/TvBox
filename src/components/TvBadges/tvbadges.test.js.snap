// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TvBadges.vue render the component 1`] = `
<div
  class="color-info-blocks"
>
  <label
    class="rating"
    title="Ratings"
  >
    â˜… 8.5
  </label>
  <label
    class="year"
    title="Premiered on"
  >
    2011
  </label>
  <label
    class="language"
    title="Language"
  >
    English
  </label>
  <label
    class="runtime"
    title="Runtime"
  >
    60 mins
  </label>
</div>
`;
